### PR TITLE
fix: ignore disk_size when disk_autoresize is true

### PR DESCRIPTION
**Pull request by HeshamMeneisi**
## Pull Request Submission Checklist

Please confirm that you have done the following before requesting reviews:

- [x] I have confirmed that the PR type is appropriate for the change I am making according to the [Honest Pull Request and Commit Message Naming Conventions](https://www.notion.so/honestbank/Pull-Request-and-Commit-Message-Naming-Conventions-bd97f2cbb34c4c73b1ff3a3e384b850c).
- [x] I have typed an adequate description that explains **why** I am making this change.
- [x] I have installed and run standard pre-commit hooks that lints and validates my code.

### Description

* Ignore `settings_disk_size` when `settings_disk_autoresize` is set to true to avoid failing to create a read-replica after the master instance has been auto-scaled up.

- - -
This change is [Reviewable](https://reviewable.io/reviews/honestbank/terraform-gcp-sql/36)

### CODE REVIEW

File: modules/google_sql_database_instance/main.tf [hunk 1]
```diff
@@ -69,7 +69,7 @@ resource "google_sql_database_instance" "instance" {
     disk_autoresize             = var.settings_disk_autoresize
     disk_autoresize_limit       = var.settings_disk_autoresize_limit
     disk_type                   = var.settings_disk_type
-    disk_size                   = var.settings_disk_size
+    disk_size                   = var.settings_disk_autoresize ? null : var.settings_disk_size
```
Comment (sunilhonest) [inline modules/google_sql_database_instance/main.tf]: What should be starting value of disk? You should have the default value of disk as 10GB and ignore the changes to the attribute disk_size in the life_cycle hooks.

**Review by sunilhonest — CHANGES_REQUESTED** [review]

File: modules/google_sql_database_instance/main.tf (hunk 1 above)
Comment (jai) [reply modules/google_sql_database_instance/main.tf]: Why would we ignore the disk size @sunilhonest ?

**Review by jai — COMMENTED** [review]
Do we need to update tests/check for this case?

File: modules/google_sql_database_instance/main.tf (hunk 1 above)
Comment (HeshamMeneisi) [reply modules/google_sql_database_instance/main.tf]: `life_cycle` already ignores updates to `disk_size`. We don't need an initial value, it's [optional](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance) and will default to the minimum size (10GB).
> [disk_size](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance#disk_size) - (Optional) The size of data disk, in GB. Size of a running instance cannot be reduced but can be increased. The minimum value is 10GB.

 If you think there's use-case where we need a manual initial size bigger than 10GB, we can only ignore it for read-replicas.

Comment (HeshamMeneisi) [reply modules/google_sql_database_instance/main.tf]: Updated to ignore`settings_disk_size` only for the read replica to allow a custom initial size for the master instance.

**Comment by HeshamMeneisi** [comment]
> Do we need to update tests/check for this case?

@jai 

The only test for this case is a scenario:

* Create a DB instance with no read-replica
* Scale it up over 10GB
* Enable read-replica

Quite complicated and not really worth it just for this case IMO.

File: modules/google_sql_database_instance/main.tf (hunk 1 above)
Comment (sunilhonest) [reply modules/google_sql_database_instance/main.tf]: @jai We have `disk_autoresize` enabled and hence GCP increases the disk size. TF apply will try to reset it to value passed(ex: 10 GB in our case) and hence we should ignore it in the life_cycle.

**Review by olli-honest — APPROVED** [review]

**Review by sunilhonest — APPROVED** [review]

**Pull request merged**